users-research/netflix/README.md

```diff
@@ -0,0 +1,248 @@
+  When discussing their use case for it, they mentioned that they could achieve
+  similar functionality differently (e.g. by emitting/listening on a 'error'
+  event), and that the ability to add catch handlers _asynchronously_ was only a
+  matter of user experience.
+
+- A significant number of respondents seem to consider that exiting the Node.js
+  process on an unhandled promise rejection is a desirable behavior.
+
+- Most of the respondents that add an `'unhandledRejection'` event listener use
+  it to log and exit the process.
+
+- However, most respondents seem not to use it, and a non-negligible part of
+  them were not aware that the `'unhandledRejection'` event exists.
+
+- More respondents seem to be adding a listener for the `uncaughtException`
+  event, and those handlers also almost always log messages and exit the
+  process. However, a non-negligible number of respondents are not aware that
+  this event exists.
+
+- No respondent seems to consider that exiting the Node.js process once a
+  promise with an unhandled rejection is garbage collected is a desirable
+  behavior.
+
+## Interpretation of survey results and interviews
+
+This section presents _our interpretation_ of the data that was collected and
+the four interviews we had with a selected sample of respondents.
+
+### Promises are used significantly at Netflix
+
+It is clear that a lot of JavaScript developers use promises at Netflix,
+including when writing Node.js applications. It also seems like a significant
+number of users are at least considering to use async/await.
+
+### Promises are confusing for users
+
+However, it's also clear that promises are confusing, even for developers who
+have a lot of experience with them. Every single one-on-one conversation we
+had with promises users highlighted at least one area of confusion with regards
+to how different JS runtimes implement promises and how they handle their
+errors.
+
+### Developers don't use static analysis tools to validate promises usage
+
+So far, we have seen that a significant number of users do not rely on static
+analysis tools to help them validate their usage of promises. As a result, it
+seems they _exclusively rely on runtime checks_ to perform this validation.
+
+### Exiting Node.js process on unhandled rejection seems to be considered as valid behavior
+
+A lot of respondents seem to be comfortable with Node.js exiting when a promise
+rejection is unhandled. The motivation for this behavior includes having better
+visibility on potentially hidden errors, and ensuring the process' state is
+consistent.
+
+The fact that the process exits synchronously or not does not seem to matter
+much for users.
+
+Some users are not comfortable with having the runtime dictate the behavior of
+the process when an unhandled rejection occurs, but they seem to be fine with
+being able to override the default behavior.
+
+### Exiting on garbage collection doesn't seem to be considered as a desirable behavior
```

Comment:
Just context: 

Exiting on garbage collection is only done because removing the ability to attach catch handlers asynchronously isn't possible given our data. If we could theoretically get everyone to agree to not use code with async catch handlers then it would be great - but I'm afraid that ship has sailed and it would meet a lot of resistance in core anyway.

Reply:
Where is the data on "exit on GC" from the survey? I didn't see any questions referring to it

Reply:
> Where is the data on "exit on GC" from the survey? I didn't see any questions referring to it

In [the survey's UI](https://docs.google.com/forms/d/1lM4TdTJcRbW2CDZw7Yj4Su75joGvK6DIkD95CenVRmk/viewanalytics), if you look at the question entitled:

> When a rejected promise doesn't have a catch handler, what would you like the behavior of your runtime to be?

You'll see on the right that there is a set of responses. The first responses down to "I don't know" were the responses that were pre-populated by the author of the survey (me). The rest are responses that were written by respondents when they chose the "Other" response.

From there you can see that the answer "waiting for that rejected promise to be garbage collected, and then exit" was available, but no respondent chose it.

Does that clarify things?

Reply:
It does - though that's not the behaviour anyone would want (me included). The behaviour I'd _actually_ want is for unhandled rejections to terminate immediately - the problem is that we can't tell if a promise rejection will never be handled so it can't be implemented.

Reply:
@benjamingr 

Also:

>  given our data

What data are you referring to?